be/src/olap/rowset/beta_rowset_writer.h

```diff
@@ -69,6 +69,7 @@ class BetaRowsetWriter : public RowsetWriter {
     RowsetTypePB type() const override { return RowsetTypePB::BETA_ROWSET; }
 
     Status get_segment_num_rows(std::vector<uint32_t>* segment_num_rows) const override {
+        std::shared_lock l(_lock);
```

Comment:
use lock_guard or change `_lock` to shared_mutex?

Reply:
fixed